product-service/src/services/productService.ts

```diff
@@ -0,0 +1,37 @@
+import { Product } from '../models/Product';
+import { executeQuery, executeTransactionQuery } from './db/pg-client';
+import { validateProductSchema } from './utils/validator.utils';
+
+export const getAllProducts = async (): Promise<Array<Product>> => {
+  const query = `select * from products inner join stocks on id = product_id where count > 0`;
+
+  return executeQuery<Product>(query);
+};
+
+export const getProductById = async (id: string): Promise<Product> => {
+    const query = `select * from products inner join stocks on id = product_id where id = '${id}'`;
+
+    const [product] = await executeQuery<Product>(query);
+
+    if (!product) {
+      throw new Error();
+    }
+
+    return product;
+  };
+
+export async function createProductInDB(product: Product): Promise<Product> {
+  validateProductSchema(product);
+
+  const { title, description, price, count } = product;
+
+  const productQuery =
+    `insert into products (title, description, price) values ('${title}', '${description}', '${price.toString()}') returning *`;
```

Comment:
this is anti-pattern. Better to use Parameterized query. You can find examples here -  https://node-postgres.com/features/queries **Parameterized query** section.